### PR TITLE
Add validation

### PR DESCRIPTION
**Pull request by mphprogrammer**

### CODE REVIEW

**Comment by mphprogrammer** [comment]
@saneshark Re: Exceptions - that's fine, I just wanted to highlight that these were contextual errors, rather than a validation error on a specific model, but no biggie. Since I don't know the implementation of attachments - how would the uploads be broken?

**Comment by saneshark** [comment]
Attachments requires

{ Message: { body: ... }, uploads: [...] }

Whereas everything else is just { body: ... }

File: spec/support/vcr_cassettes/sm_client/messages/creates/cannot_send_draft_as_reply.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (saneshark) [inline spec/support/vcr_cassettes/sm_client/messages/creates/cannot_send_draft_as_reply.yml]: I don't think we should need a yml file for these, these should be tested as validations and validations only.

Comment (saneshark) [inline spec/support/vcr_cassettes/sm_client/messages/creates/cannot_send_draft_as_reply.yml]: Actually, wait... nm, i guess this tests the call to history. This is OK. ignore previous comment.

**Comment by saneshark** [comment]
I think this is almost good to go. As a sanity check I would prefer if we re-run VCR just to see if anything is broken after this, but if its not, don't bother committing those changes, instead check the files out from master again if that makes sense. I can run these for you if you want, it's a little annoying still and I know how to update the specs when and if necessary.

**Comment by mphprogrammer** [comment]
@saneshark I rewrote and simplified. Ran specs for sending message w/attachments and replies w/attachments - it ran ok. If you'd like and can rerun specs w/out attachments too.

**Comment by saneshark** [comment]
When you say reran specs, you deleted the VCR cassettes and reran or you ran with existing cassettes? I meant return and rerecord cassettes, committing those new cassettes only if necessary.

**Comment by mphprogrammer** [comment]
@saneshark I deleted the ymls and rerecorded.

**Comment by mphprogrammer** [comment]
OK - merging.

**Pull request merged**